Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -16,59 +16,212 @@ import Foundation
 import CommonCrypto
 #endif
 
+/// This represents the possible option using in validating a claim
+public struct ValidationOption: OptionSet  {
+    public let rawValue: Int
+    
+    public init(rawValue: Int) {
+        self.rawValue = rawValue
+    }
+    
+    ///  This represents the default validation checks in-order using to validate a claim
+    public static let all: ValidationOption = [
+        .validateIssuer,
+        .validateAudience,
+        .validateScheme,
+        .validateAlgorithm,
+        .validateExpirationTime,
+        .validateIssuedAtTime,
+        .validateNonce,
+        .validateAuthTime,
+        .validateMaxAge,
+        .validateSubject
+    ]
+    
+    /// Validate the Issuer claim of the token
+    static let validateIssuer = ValidationOption(rawValue: 1 << 0)
+    /// Validate the Audience claim
+    static let validateAudience = ValidationOption(rawValue: 1 << 1)
+    /// Validate the Issuer scheme  of the URL.
+    static let validateScheme = ValidationOption(rawValue: 1 << 2)
+    /// Validate the signing algorithm used to sign this JWT token.
+    static let validateAlgorithm = ValidationOption(rawValue: 1 << 3)
+    /// Validate the Expiration Time. exp claim
+    static let validateExpirationTime = ValidationOption(rawValue: 1 << 4)
+    /// Validate the date this token was issued. iat claim
+    static let validateIssuedAtTime = ValidationOption(rawValue: 1 << 5)
+    /// Validate the nonce claim
+    static let validateNonce = ValidationOption(rawValue: 1 << 6)
+    /// Validate the date at which authentication occurred.
+    static let validateAuthTime = ValidationOption(rawValue: 1 << 7)
+    /// Validate the maximum age the token should support when authenticating.
+    static let validateMaxAge = ValidationOption(rawValue: 1 << 8)
+    /// Validate the subject of the resource, if available.
+    static let validateSubject = ValidationOption(rawValue: 1 << 9)
+}
+
 struct DefaultIDTokenValidator: IDTokenValidator {
     var issuedAtGraceInterval: TimeInterval = 300
-    
+        
     func validate(token: JWT, issuer: URL, clientId: String, context: IDTokenValidatorContext?) throws {
+        let validationOptions = context?.validationOptions ?? .all
+        
```

Comment:
I don't think we need to break all of these into separate functions. If anything, it would just add to the symbol table overhead.

I'm thinking something like this:

```swift
        for check in checks {
            switch check {
            case .issuer:
                guard tokenIssuer.absoluteString == issuer.absoluteString
                else {
                    throw JWTError.invalidIssuer
                }
                
            case .audience:
                guard token[.audience] == clientId else {
                    throw JWTError.invalidAudience
                }
   ...
```

The nice thing is the diff for this commit will be minimal, since the code will be laid out mostly the same, just a few lines of indentation and a switch / case block.